### PR TITLE
Auto-generate program codes

### PR DESCRIPTION
**Pull request by ThiefMaster**
I left editing the templates in a single dialog for now, since it seem to be more convenient to have them all in one place (you usually only set them up once).

### CODE REVIEW

File: indico/modules/events/management/templates/program_codes.html [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+                data-reload-after
+                data-ajax-dialog></button>
+    </h3>
+    <dl class="i-data-list">
+        <dt>{% trans %}Sessions{% endtrans %}</dt>
+        <dd>
+            {% if templates.session_template %}
+                <code>{{ templates.session_template }}</code>
+                <button class="icon-play i-button i-button-small text-color borderless"
+                        style="padding: 0;"
```
Comment (mic4ael) [inline indico/modules/events/management/templates/program_codes.html]: I am wondering whether you forgot to move it to a `scss` file :P

File: indico/modules/events/management/templates/program_codes.html [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+                        data-title="{% trans %}Assign programme codes{% endtrans %}"
+                        data-ajax-dialog></button>
+            {% else %}
+                <em>{% trans %}No template defined{% endtrans %}</em>
+            {% endif %}
+        </dd>
+        <dt>{% trans %}Session blocks{% endtrans %}</dt>
+        <dd>
+            {% if templates.session_block_template %}
+                <code>{{ templates.session_block_template}}</code>
```
Comment (mic4ael) [inline indico/modules/events/management/templates/program_codes.html]: missing space before `}}`

File: indico/modules/events/management/templates/assign_program_codes.html [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+{% set has_codes = codes|selectattr('code')|any %}
+
+{% macro render_row(object, code, unused) %}
+    <tr class="i-table">
+        <td class="i-table title-column">
+            {% if unused %}
+                <i class="icon-warning semantic-text warning" data-qtip-html=".qtip-hidden-content"></i>
+                <div class="qtip-hidden-content">
+                    <div style="text-align: left;">
```
Comment (mic4ael) [inline indico/modules/events/management/templates/assign_program_codes.html]: Did you leave it on purpose here as well?

File: indico/modules/events/management/controllers/program_codes.py [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+
+class RHAssignProgramCodesContributions(RHAssignProgramCodesBase):
+    object_type = 'contributions'
+    hidden_post_field = 'contribution_id'
+
+    def _get_objects(self):
+        ids = map(int, request.form.getlist('contribution_id'))
+        return (Contribution.query
+                .with_parent(self.event)
+                .filter(Contribution.id.in_(ids) if ids else True)
```
[the remark this answers is not in the document]
Comment (ThiefMaster) [reply indico/modules/events/management/controllers/program_codes.py]: That's included in the `Event.contributions` relationship which `with_parent` uses

**Pull request merged**